### ISSUE TITLE
chore: draft release note 0.7.0

### ISSUE THREAD

**Issue by scott-martens** [jina-ai/finetuner#652]
# Release Note

This release covers Finetuner version 0.6.8, including dependencies finetuner-api 0.4.9 and finetuner-core 0.11.5.

[//]: <> (remove the phrase like "0 new features")
This release contains 2 new features, 0 performance improvements, 1 bug fixes and 2 Documentation improvements.

[//]: <> (Does it have security update? If yes then add the following lines)
It is recommended to upgrade to this version as it contains security updates.

## 💥 Security Updates

[//]: <> (If no security update please remove this session.)

### Add `val_split` to fit interface (#624)

[//]: <> (From Finetuner-API)
### Add argo runner (#189)

[//]: <> (From Finetuner-Core)
### Add support for multilingual clip (#376)

[//]: <> (Nicely introduce each feature before/after with some code snippet and results)

## 🚀 Performance

[//]: <> (Nicely introduce each performance improvement before/after with some code snippet and results)

We have made a performance improvement to the system. When running the following code:

The result is 123% faster than before.

-The load time for the system has been significantly reduced.
-The system is now more responsive and faster.
-The overall performance of the system has been improved.

### Adopt api changes (#637)

[//]: <> (From Finetuner-API)
### Allow get runs without exp name (#198)

[//]: <> (From Finetuner-Core)
### Fix build_encoding_dataset (#623)

[//]: <> (From Finetuner-API)
### Iam permissions in usage insights

[//]: <> (From Finetuner-Core)
### Log messages in evaluation callback (#419)

[//]: <> (Nicely introduce each bug fix before/after with some code snippet and results)

- The bug was causing [explain what the bug was causing].
- The new version fixes the bug and [explain what the new version does].

## 📗 Documentation Improvements

- Add before and after to text-text (#622)
- Modify hints to use blobs instead of tensor (#625)

[//]: <> (Nicely introduce each feature before/after with some code snippet and results)

## 🤟 Contributors

We would like to thank all contributors to this release:
- Wang Bo (@bwanglzu)
- Michael Günther (@guenthermi)
- Louis Milliken (@LMMilliken)
- George Mastrapas (@Unknown user)

**Comment by guenthermi** [comment]
# Release Note

This release covers Finetuner version 0.7.0, including dependencies finetuner-api 0.4.9 and finetuner-core 0.11.5.

[//]: <> (remove the phrase like "0 new features")
This release contains 4 new features, 4 bug fixes, and 4 Documentation improvements.

[//]: <> (Does it have security update? If yes then add the following lines)

## 🆕 Features

### Allow Fine-Tuning PointNet++ Models (#638)

We add a new embedding model based on the [PointNet++ model](https://proceedings.neurips.cc/paper/2017/file/d8bf84be3800d12f74d8b05e9b89836f-Paper.pdf).
This model can be used to build a search system for 3D meshes.
You can start a fine-tuning run of the model by setting teh model parameter to `pointnet++`:

```python
import finetuner

finetuner.login()

run = finetuner.fit(
    model='pointnet++',
    train_data='finetuner/modelnet40-train',
    epochs=10,
    batch_size=64,
    learning_rate= 5e-4,
    loss='TripletMarginLoss',
    device='cuda',
)
```

We also prepared a [tutorial](https://colab.research.google.com/drive/1lIMDFkUVsWMshU-akJ_hwzBfJ37zLFzU?usp=sharing) with detailed information about how to use the model and prepare 3D mesh data for the finetuning fine-tuning.

### Add `val_split` to fit interface (#624)

To make it easier to evaluate models, we add a new parameter to the fit function called `val_split`.
This enables users to automatically split the training data into a training dataset and a validation dataset which is used to calculate an evaluation loss.
For example, the following call will automatically hold out 20% of the data for the validation:
```python
run = finetuner.fit(
    model='efficientnet_b0',
    train_data=train_data,
    val_split=0.2
)
```

[//]: <> (From Finetuner-Core)

## Add recall and f1 score to evaluation metrics.

If you are using the evaluation callback, you can now see the evaluation results with respect to two new metrics: `recall_at_k` and `f1_score_at_k`.

## Evaluate zero-shot model via the evaluation callback and include this into the logs.

Previously, the evaluation callback only evaluated the models after each fine-tuning epoch.
Now, the evaluation metrics are also calculated before the fine-tuning and the logs display the metrics after each epoch together with the evaluation results calculated for the zero shot model:

```
DEBUG    Finetuning took 0 days, 0 hours 5 minutes and 39 seconds                                         __main__.py:197
INFO     Metric: 'pointnet++_precision_at_k' before fine-tuning:  0.56533 after fine-tuning: 0.81100      __main__.py:210
INFO     Metric: 'pointnet++_recall_at_k' before fine-tuning:  0.15467 after fine-tuning: 0.24175         __main__.py:210
INFO     Metric: 'pointnet++_f1_score_at_k' before fine-tuning:  0.23209 after fine-tuning: 0.34774       __main__.py:210
INFO     Metric: 'pointnet++_hit_at_k' before fine-tuning:  0.95667 after fine-tuning: 0.95333            __main__.py:210
INFO     Metric: 'pointnet++_average_precision' before fine-tuning:  0.71027 after fine-tuning: 0.85515   __main__.py:210
INFO     Metric: 'pointnet++_reciprocal_rank' before fine-tuning:  0.79103 after fine-tuning: 0.89103     __main__.py:210
INFO     Metric: 'pointnet++_dcg_at_k' before fine-tuning:  4.71826 after fine-tuning: 6.41999 
```

[//]: <> (Nicely introduce each feature before/after with some code snippet and results)


## 🚀 Performance



[//]: <> (Nicely introduce each performance improvement before/after with some code snippet and results)

We have made a performance improvement to the system. When running the following code:

The result is 123% faster than before.

-The load time for the system has been significantly reduced.
-The system is now more responsive and faster.
-The overall performance of the system has been improved.

## ⚙ Refactoring

### Change default experiment_name from current working dir to default. (#637)

Previously, the experiment was named after the current folder if no other experiment name was provided.
Now, generic name `default` is used instead.
Please note, that you can not create two runs with the same name in the same experiment.


### Deprecate cpu parameter (#631)

The `cpu` parameter of the `fit` function is deprecated. Please use instead the new `device` parameter which can be set to `cuda` or `cpu`:

```diff
run = finetuner.fit(
    model='efficientnet_b0',
    train_data=train_data,
-   cpu=True
+   device='cpu'
)
```

### Deprecate notebook_login function (#631)

Beside the ordinary `login()` function there used to be a `notebook_login()` function which needd to be used in Jupyter notebooks instead of `login()`.
Since a while, you don't need to use this function anymore.
Instead `login()` works correctly in notebooks.
Since this release the old `notebook_login()` function is removed.

### Adopt API changes (#637)

We did some small changes in the interface of our API.
In this turn, we changed the `list_runs` and `list_experiments` function.
Those functions, have two new optional arguments: `page` and `size`.
You can set `size` to only retrieve a certain number of runs / experiments.
By increasing the `page` argument one by one, you can retrieve the runs / experiments in batches with as many elemets as defined via `size`.


## 🐞 Bug Fixes

### Fix build_encoding_dataset (#623)

If you wanted to encode a list of texts or images with the CLIP model since might have worked only by constructing a `DocumentArray` object.
If you where passing a Python list to the `finetuner.encode()` method, you may received an error since Finetuner was not able to correctly detect the modality of the data.
The should not finally work:

```
finetuner.encode(clip_text_model, ['some text to encode'])
```

[//]: <> (From Finetuner-Core)

### Adjust `num_items_per_class` if necessary (#441)

The `finetuner.fit` function has a `num_items_per_class` parameter which determine how many items per class should be put into a batch during the training.
Unfortunately, it is not possible to freely set this parameter for evary `batch_size` ane every dataset, which could led to errors during the training.
To prevent those errors, we decided to automatically adjust `num_items_per_class` if it is not compatible with your configuration.
Thereby, Finetuner will always try to find a suitable parameter close to the one you selected.
In this way, you only will receive a warning that the parameter has been adjusted.

### Set default freeze to false (#436)

If you used the `build_model` function to construct a CNN model, the model was automatically build with `freeze=True`. In this case, Finetuner also adds a projection head to the function.
To avoid this, `freeze` is now set to `False` by default.

### Log messages in evaluation callback (#419)

Previously, some logging messages of the evaluation callback where overwritten by progress bars.
Now, this should not happen anymore.

[//]: <> (Nicely introduce each bug fix before/after with some code snippet and results)

- The bug was causing [explain what the bug was causing].
- The new version fixes the bug and [explain what the new version does].

## 📗 Documentation Improvements

### Re-write the README (#638, #643)

We re-write the README to be more compact and include results of PointNet++ fine-tuning

### Rewrite M-CLIP notebook to use German fashion dataset. (#643)

Our M-CLIP tutorial is now based on a German fashion dataset, which represents a more realistic application scenario.


- Add before and after to text-text (#622)
- Modify hints to use blobs instead of tensor (#625)

### Add before and after examples in the tutorials (#622)

In our tutorials, we include some examples to let you compare results before and after fine-tuning.

### Restructuring and enriching our documentation as a whole (#643)

We did a large re-writing of our documentation pages.
Some of the most significant changes is that we include a section with advanced topics like "Negative Mining" and add a sub section about inference to the Walkthrough section. 
We also improve the appearance of our developer reference. 


[//]: <> (Nicely introduce each feature before/after with some code snippet and results)


## 🤟 Contributors

We would like to thank all contributors to this release:
- Wang Bo (@bwanglzu)
- Michael Günther (@guenthermi)
- Louis Milliken (@LMMilliken)
- George Mastrapas (@gmastrapas)

**Comment by scott-martens** [comment]
# Release Note

This release covers Finetuner version 0.7.0, including dependencies finetuner-api 0.4.9 and finetuner-core 0.12.2.

This release contains 4 new features, 4 bug fixes, and 4 documentation improvements.

## 🆕 Features

### Allow Fine-Tuning for PointNet++ Models ([#638](https://github.com/jina-ai/finetuner/pull/638))

We have added a new embedding model based on the [PointNet++ model](https://proceedings.neurips.cc/paper/2017/file/d8bf84be3800d12f74d8b05e9b89836f-Paper.pdf). You can use this model for text-to-3D mesh applications. To fine-tune it, set the `model` parameter of the `fit` function to `pointnet++`:

```python
import finetuner

finetuner.login()

run = finetuner.fit(
    model='pointnet++',
    train_data='finetuner/modelnet40-train',
    epochs=10,
    batch_size=64,
    learning_rate= 5e-4,
    loss='TripletMarginLoss',
    device='cuda',
)
```

We have also prepared a [tutorial](https://colab.research.google.com/drive/1lIMDFkUVsWMshU-akJ_hwzBfJ37zLFzU?usp=sharing) with detailed information about how to use the model and prepare 3D mesh data for Finetuner.

### Add `val_split` to fit interface ([#624](https://github.com/jina-ai/finetuner/pull/624))

To make it easier to evaluate models, we have added the `val_split` parameter to the fit function. Using this parameter automatically splits the training data into a training dataset and a validation dataset, used to calculate loss. For example, the following call will automatically hold out 20% of the data for validation:
```python
run = finetuner.fit(
    model='efficientnet_b0',
    train_data=train_data,
    val_split=0.2
)
```

[//]: <> (From Finetuner-Core)

### Add recall and F1 score to evaluation metrics

If you are using the evaluation callback, evaluation results now include two new metrics: `recall_at_k` and `f1_score_at_k`.

### Evaluation callback evaluates zero-shot models and reports this in the logs

Previously, the evaluation callback only evaluated models after each fine-tuning epoch. Now, evaluation metrics are also calculated before any fine-tuning and the log displays the metrics after each epoch, together with the pre-fine-tuning (i.e. zero-shot) evaluation results:

```
DEBUG    Finetuning took 0 days, 0 hours 5 minutes and 39 seconds                                         __main__.py:197
INFO     Metric: 'pointnet++_precision_at_k' before fine-tuning:  0.56533 after fine-tuning: 0.81100      __main__.py:210
INFO     Metric: 'pointnet++_recall_at_k' before fine-tuning:  0.15467 after fine-tuning: 0.24175         __main__.py:210
INFO     Metric: 'pointnet++_f1_score_at_k' before fine-tuning:  0.23209 after fine-tuning: 0.34774       __main__.py:210
INFO     Metric: 'pointnet++_hit_at_k' before fine-tuning:  0.95667 after fine-tuning: 0.95333            __main__.py:210
INFO     Metric: 'pointnet++_average_precision' before fine-tuning:  0.71027 after fine-tuning: 0.85515   __main__.py:210
INFO     Metric: 'pointnet++_reciprocal_rank' before fine-tuning:  0.79103 after fine-tuning: 0.89103     __main__.py:210
INFO     Metric: 'pointnet++_dcg_at_k' before fine-tuning:  4.71826 after fine-tuning: 6.41999 
```

## ⚙ Refactoring

### Drop support for python version 3.7

Due to a new dependency in the stubs package of finetuner-core that is not supported by python 3.7, finetuner now requires python 3.8 or higher.


### Change default experiment_name from current working dir to default ([#637](https://github.com/jina-ai/finetuner/pull/637))

Previously, Finetuner named experiments after the current folder, if no other name was provided. Now, the generic name `default` is used instead. Please note, that you can not create two runs with the same name in the same experiment, and we recommend always giving experiments explicit names.

### Add `page` and `size` parameters to `list_runs` and `list_experiments` functions ([#637](https://github.com/jina-ai/finetuner/pull/637))

We have added two new optional arguments to the `list_runs` and `list_experiments` functions: `page` and `size`. You can set `size` so that you retrieve no more than a specific number of runs or experiments. To retrieve more, set the `page` argument to progressively higher values. You can use these parameters to paginate the list of runs and experiments, which can become quite long.


### Deprecate `cpu` parameter and `notebook_login` function ([#631](https://github.com/jina-ai/finetuner/pull/631))

The `cpu` parameter of the `fit` function is deprecated. Instead, use the `device` parameter which can be set to `cuda` or `cpu`:

```diff
run = finetuner.fit(
    model='efficientnet_b0',
    train_data=train_data,
-   cpu=True
+   device='cpu'
)
```
Additionally, we have deprecated the `notebook_login()` function. This was necessary to specifically to support Jupyter notebooks, but `login()` now works correctly in notebooks by itself. We have removed the `notebook_login()` function and you must update any code that calls it.

## 🐞 Bug Fixes

### Fix build_encoding_dataset ([#623](https://github.com/jina-ai/finetuner/pull/623))

Previously, if you passed Finetuner a list of strings to encode, it would fail because it was unable to determine if this was a list of texts or a list of URIs to images or other objects. You would need explicit state the type in a `DocumentArray` object. Finetuner can now automatically detect the data type and handle it correctly.

```
finetuner.encode(clip_text_model, ['some text to encode'])
```

[//]: <> (From Finetuner-Core)

### Adjust `num_items_per_class` if necessary

The `finetuner.fit` function has a `num_items_per_class` parameter which determines how many items per class should be put into a batch during training. Unfortunately, it is not possible to freely set this parameter for every `batch_size` and every dataset, which could lead to errors during the training. Finetuner will now automatically adjust `num_items_per_class` if the one provided by the user is not compatible with the rest of the batch configuration.
 
Finetuner will try to find a parameter value close to the one you provided. You will only receive a warning that the parameter has been adjusted, and training will continue.

### Set default freeze to false

Until now, the `build_model` function would automatically set the parameter `freeze=True` when constructing a CNN model. Finetuner would also add a projection head to the function. To avoid this, `freeze` is now set to `False` by default.

### Log messages in evaluation callback

Previously, some logging messages from the evaluation callback were overwritten by progress bars in some cases. This should no longer occur.

## 📗 Documentation Improvements

### Re-write the README ([#638](https://github.com/jina-ai/finetuner/pull/638), [#643](https://github.com/jina-ai/finetuner/pull/643))

We have re-written the README file to be more concise and to include results from PointNet++ fine-tuning

### Rewrite the M-CLIP notebook to use the German Fashion12k dataset ([#643](https://github.com/jina-ai/finetuner/pull/643))

Our M-CLIP tutorial now uses the [German Fashion12k dataset](https://jina.ai/news/improving-search-quality-non-english-queries-fine-tuned-multilingual-clip-models/), which represents a more realistic application scenario.

### Add before and after examples in the tutorials ([#622](https://github.com/jina-ai/finetuner/pull/623))

In our tutorials, we now include some examples that let you compare results before and after fine-tuning.

### Restructuring and enriching our documentation as a whole ([#643](https://github.com/jina-ai/finetuner/pull/643))

We have performed a substantial re-write of our documentation pages. This includes new advanced topics like "Negative Mining" and information about inference in the __Walkthrough__ section. We also improved the developer reference.

## 🤟 Contributors

We would like to thank all contributors to this release:
- Wang Bo ([@bwanglzu](https://github.com/bwanglzu))
- Michael Günther ([@guenthermi](https://github.com/guenthermi))
- Louis Milliken ([@LMMilliken](https://github.com/LMMilliken))
- George Mastrapas ([@gmastrapas](https://github.com/gmastrapas))
- Scott Martens ([@scott-martens](https://github.com/scott-martens))
